src/tests/functionspace/test_pointcloud_haloexchange_3PE.cc

```diff
@@ -0,0 +1,134 @@
+/*
+ * (C) Copyright 2023 ECMWF
+ * (C) Crown Copyright 2023 Met Office
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ */
+
+
+#include <vector>
+
+#include "atlas/array.h"
+#include "atlas/field.h"
+#include "atlas/functionspace/PointCloud.h"
+#include "atlas/option.h"
+#include "atlas/parallel/mpi/mpi.h"
+
+#include "tests/AtlasTestEnvironment.h"
+
+using namespace eckit;
+using namespace atlas::functionspace;
+using namespace atlas::util;
+
+namespace atlas {
+namespace test {
+
+
+CASE("test_halo_exchange_01") {
+
+  //
+  //     ++ point order ++
+  //
+  //       PE 0     PE 1
+  //     _________________
+  //    | 0    1 || 0   1 |
+  //    | 3    2 || 3   2 |
+  //    ------------------
+  //    ------------------
+  //    | 0    1    2   3 |    
+  //    | 7    6    5   4 |
+  //    ------------------
+  //           PE 2
+  //
+  // 'own points': clockwise starting from top left
+  //
+
+  const eckit::mpi::Comm& comm = atlas::mpi::comm();
+  std::size_t mpi_rank = comm.rank();
+  const std::size_t no_mpi_ranks = comm.size();
```

Comment:
Cosmetic change: Please use "mpi_size" as name.

Reply:
Actually, these lines were not used. I removed them.